internal/services/integrationtesting/cert_test.go

```diff
@@ -137,24 +137,40 @@ func TestCertRotation(t *testing.T) {
 		server.WithDashboardAPI(util.HTTPServerConfig{HTTPEnabled: false}),
 		server.WithMetricsAPI(util.HTTPServerConfig{HTTPEnabled: false}),
 		server.WithDispatchServer(util.GRPCServerConfig{Enabled: false}),
-		server.SetMiddlewareModification([]server.MiddlewareModification{
+		server.SetUnaryMiddlewareModification([]server.MiddlewareModification[grpc.UnaryServerInterceptor]{
 			{
 				Operation: server.OperationReplaceAllUnsafe,
-				Middlewares: []server.ReferenceableMiddleware{
+				Middlewares: []server.ReferenceableMiddleware[grpc.UnaryServerInterceptor]{
 					{
-						Name:                "datastore",
-						UnaryMiddleware:     datastoremw.UnaryServerInterceptor(ds),
-						StreamingMiddleware: datastoremw.StreamServerInterceptor(ds),
+						Name:       "datastore",
+						Middleware: datastoremw.UnaryServerInterceptor(ds),
 					},
 					{
-						Name:                "consistency",
-						UnaryMiddleware:     consistency.UnaryServerInterceptor(),
-						StreamingMiddleware: consistency.StreamServerInterceptor(),
+						Name:       "consistency",
+						Middleware: consistency.UnaryServerInterceptor(),
 					},
 					{
-						Name:                "servicespecific",
-						UnaryMiddleware:     servicespecific.UnaryServerInterceptor,
-						StreamingMiddleware: servicespecific.StreamServerInterceptor,
+						Name:       "servicespecific",
+						Middleware: servicespecific.UnaryServerInterceptor,
+					},
+				},
+			},
+		}),
+		server.SetStreamingMiddlewareModification([]server.MiddlewareModification[grpc.StreamServerInterceptor]{
+			{
+				Operation: server.OperationReplaceAllUnsafe,
+				Middlewares: []server.ReferenceableMiddleware[grpc.StreamServerInterceptor]{
+					{
```

Comment:
I wonder if we want to verify that the same interceptors are set on both unary and streaming, to ensure that we don't miss one?

Reply:
If we went back to having unary/streaming coupled into one, then you could do the validation at the aggregation instead:

```go
Middleware {
 MiddelwareID
 UnaryInterceptor
 StreamingInterceptor
}

MiddlewareChain {
   []Middleware
   UnaryOrder []MiddlewareID
   StreamingOrder []MiddlewareID
}
```

A validate method or a constructor could assert that every non-nil interceptor is somewhere in the order list?

Not sure this adds that much value, though.

Reply:
I'm ve gone with an approach that asserts in tests that the order executed is what we expect, see https://github.com/authzed/spicedb/pull/1395

Reply:
oh I see what you say 👍🏻